src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,48 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.BulldozerProducer;
+import core.mate.academy.ExcavatorProducer;
+import core.mate.academy.TruckProducer;
+import core.mate.academy.model.Machine;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    private static final String BULLDOZER = "Bulldozer";
+    private static final String EXCAVATOR = "Excavator";
+    private static final String TRUCK = "Truck";
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        MachineProducer<? extends Machine> machineProducer;
+        switch (type.getSimpleName()) {
+            case BULLDOZER:
+                machineProducer = new BulldozerProducer();
```

Comment:
make your producers class level fields, to avoid creating a new instance of each producer when calling this method